### PR TITLE
Add a pipeline_id column to buildsets

### PR DESCRIPTION
**Pull request by rajdeepbharati**
# Contributor Checklist:

* [x] I have updated the unit tests
* [ ] I have created a file in the `master/buildbot/newsfragments` directory (and read the `README.txt` in that directory)
* [ ] I have updated the appropriate documentation

### CODE REVIEW

**Comment by sa2ajj** [comment]
Could you please add some docs/release notes that'd describe what the goal is?  The PR description nor the code tell much...

**Comment by rajdeepbharati** [comment]
@sa2ajj The end goal is to fix this issue https://github.com/buildbot/buildbot/issues/3995
This is still in progress. I'll update the description and add docs once I finish the work.

**Comment by tardyp** [comment]
I agree with @sa2ajj that this needs to come with very strong architecture design on how this will work eventually.
I am afraid  we are out of time for gsoc2019 to start such large refactor task.

I am not telling that it is a bad idea, but before commiting into db change, we should really be sure this is the right way to go (or we should do it in a different branch)

**Comment by rajdeepbharati** [comment]
@tardyp Should I try having the pipelineid same as buildid, or different?

**Comment by tardyp** [comment]
> @tardyp Should I try having the pipelineid same as buildid, or different?

looks like a good idea

**Comment by rajdeepbharati** [comment]
@tardyp Ok, thanks. But now I'm facing a different issue which seems to be due to the previous migration that made committer not null (055): https://paste.macports.org/c6305c433160

File: master/buildbot/test/unit/test_db_migrate_versions_056_add_buildsets_pipeline_id.py [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+
+            # q = sa.select([buildsets.c.submitted_at, buildsets.c.pipeline_id])
+            # num_rows = 0
+            # for row in conn.execute(q):
+            #     # verify that the default value was set correctly
+            #     self.assertEqual(row.pipeline_id, 100)
+            #     num_rows += 1
+            # self.assertEqual(num_rows, 1)
+
+        return self.do_test_migration(54, 55, setup_thd, verify_thd)
```
Comment (tardyp) [inline master/buildbot/test/unit/test_db_migrate_versions_056_add_buildsets_pipeline_id.py]: you do a migration here from version 54 to version 55. So it runs the 055 migration. You should write:
```suggestion
        return self.do_test_migration(55, 56, setup_thd, verify_thd)
```

Comment (rajdeepbharati) [reply master/buildbot/test/unit/test_db_migrate_versions_056_add_buildsets_pipeline_id.py]: Oops. Sorry for doing such a trivial mistake

**Pull request closed without merging**